src/main/java/org/jabref/gui/preftabs/PreviewPrefsTab.java

```diff
@@ -126,8 +131,19 @@ private void setupLogic() {
                 DefaultTaskExecutor.runInJavaFXThread(() -> {
 
                     PreviewPanel testPane = new PreviewPanel(null, null, Globals.getKeyPrefs(), Globals.prefs.getPreviewPreferences(), dialogService);
-                    testPane.setFixedLayout(layout.getText());
-                    testPane.setEntry(TestEntry.getTestEntry());
+                    if (chosen.isSelectionEmpty()) {
+                        testPane.setFixedLayout(layout.getText());
+                        testPane.setEntry(TestEntry.getTestEntry());
+                    }
+                    else {
+                        int indexStyle = chosen.getSelectedIndex();
+                        PreviewPreferences p = Globals.prefs.getPreviewPreferences();
+                        p = new PreviewPreferences(p.getPreviewCycle(),indexStyle,p.getPreviewPanelDividerPosition(),p.isPreviewPanelEnabled(), p.getPreviewStyle(),p.getPreviewStyleDefault());
+
+                        testPane = new PreviewPanel(Globals.basePanel, new BibDatabaseContext(), Globals.getKeyPrefs(), p, dialogService);
```

Comment:
Do you really need the base panel here? If you look at the line 133 where the PreviewPanel is initialized for the first time, there is null passed for the basePanel and the bibdatabasecontex.   So I assume you don't need the Globals.basepanel hack here.

Reply:
Yes, I need the basepanel because in the PreviewPanel.java file checks if basePanel is present in order to get the citation otherwise it doesn't work. 

At first I initialized it with null because if no citation style is selected generates a default citation. So that's why I did it that way, the other thing I could try is to write the same logic there without calling the function updateLayout in PreviewPanel but it'll end up with duplicated code. What do you think is the best approach?

Reply:
Ah yes, I now did take a closer look at the code here.  Having a dependency on the BasePanel is not nice. Well, but looking at the code and trying out some things I see no other way as the CitationStyleCache is configured on the basePanel.
However, you can use the method here    `JabRefGUI.getMainFrame().getCurrentBasePanel() `. so you don't need that dependency on base panel in globals.

Reply:
Thank you, I remove it from Globals and use getCurrentBasePanel instead and it worked.